src/main/antlr4/GobraParser.g4

```diff
@@ -0,0 +1,428 @@
+
+make: MAKE L_PAREN type_ (COMMA expressionList)? R_PAREN;
+
+new_: NEW L_PAREN type_ R_PAREN;
+
+// Added ghost statements
+statement:
+    ghostStatement
+    | packageStmt
+    | applyStmt
+	| declaration
+	| labeledStmt
+	| simpleStmt
+	| goStmt
+	| returnStmt
+	| breakStmt
+	| continueStmt
+	| gotoStmt
+	| fallthroughStmt
+	| block
+	| ifStmt
+	| switchStmt
+	| selectStmt
+	| specForStmt
+	| deferStmt;
+
+applyStmt: APPLY expression;
+
+packageStmt: PACKAGE expression block?;
+
+specForStmt: loopSpec forStmt;
+
+loopSpec: (INV expression eos)* (DEC terminationMeasure eos)?;
+
+terminationMeasure: expressionList? (IF expression)?;
+
+// Added true, false as literals
+basicLit:
+    TRUE
+	| FALSE
+	| NIL_LIT
+	| integer
+	| string_
+	| FLOAT_LIT
+	| IMAGINARY_LIT
+	| RUNE_LIT;
+
+// Added ghostPrimaryExprs
+// Fixed arguments matching on the next line
+primaryExpr:
+	operand
+	| conversion
+	| methodExpr
+	| ghostPrimaryExpr
+	| primaryExpr (
+		(DOT IDENTIFIER)
+		| index
+		| slice_
+		| seqUpdExp
+		| typeAssertion
+		| {noTerminatorBetween(1)}? arguments
+		| predConstructArgs
+	);
+
+predConstructArgs: L_PRED expressionList? COMMA? R_PRED;
+
+// Added predicate spec and method specifications
+interfaceType:
+	INTERFACE L_CURLY ((methodSpec | typeName| predicateSpec) eos)* R_CURLY;
+
+predicateSpec: PRED IDENTIFIER parameters;
+
+methodSpec:
+	{noTerminatorAfterParams(2)}? GHOST? specification IDENTIFIER parameters result
+	| GHOST? specification IDENTIFIER parameters;
+
+// Added ghostTypeLiterals
+type_: typeName | typeLit | ghostTypeLit | L_PAREN type_ R_PAREN
+        | predefined=(
+               BOOL |
+                 STRING |
+                 PERM |
+                // signed integer types
+                 RUNE |
+                 INT |
+                 INT8 |
+                 INT16 |
+                 INT32 |
+                 INT64 |
+                // unsigned integer types
+                 BYTE |
+                 UINT |
+                 UINT8 |
+                 UINT16 |
+                 UINT32 |
+                 UINT64 |
+                 UINTPTR |
+                 FLOAT32 |
+                 FLOAT64 |
+                 COMPLEX64 |
+                 COMPLEX128
+        );
+// Added pred types
+typeLit:
+	arrayType
+	| structType
+	| pointerType
+	| functionType
+	| interfaceType
+	| sliceType
+	| mapType
+	| channelType
+	| predType;
+
+predType: PRED predTypeParams;
+
+predTypeParams: L_PAREN (type_ (COMMA type_)* COMMA?)? R_PAREN;
+
+// Ditto
+literalType:
+	structType
+	| arrayType
+	| L_BRACKET ELLIPSIS R_BRACKET elementType
+	| sliceType
+	| mapType
+	| ghostTypeLit
+	| typeName;
+
+// Added Deflate
+//exprCaseClause: exprSwitchCase COLON statementList?;
+
+
+// ANTLR Grammar fixes
+
+// distinguish low,high cap
+slice_:
+	L_BRACKET (
+		low? COLON high?
+		| low? COLON high COLON cap
+	) R_BRACKET;
+
+low : expression;
+high: expression;
+cap: expression;
+
+// allow "if ; true {}"
```

Comment:
This was actually an error in the GoParser.g4, but I've since fixed it in the original repo, so this should move back to the GoParser.g4, thanks for noticing.